### PR TITLE
README: clarify the linking exception

### PR DESCRIPTION
**Pull request by ethomson**
We say that you can link libgit2 "unmodified"... "without having to release its source code". Clarify that you can modify libgit2 - but you must release _its_ source code back - and you can link libgit2 without having to release _your software's_ source code.

### CODE REVIEW

**Comment by ethomson** [comment]
Fixes #6467

File: README.md [hunk 1]
```diff
@@ -25,10 +25,11 @@ and on Git hosting providers like [GitHub](https://github.com/),
 We perform the merge every time you click "merge pull request".
 
 `libgit2` is licensed under a **very permissive license** (GPLv2 with a special
-Linking Exception).  This basically means that you can link it (unmodified)
-with any kind of software without having to release its source code.
-Additionally, the example code has been released to the public domain (see the
-[separate license](examples/COPYING) for more information).
+Linking Exception). The GPLv2 requires you to make changes you make _to libgit2
+itself_ available publicly, but the linking exception means that you can link
```
Comment (carlosmn) [inline README.md:29]: The GPL does not require publishing modifications publicly. it requires you to provide source code to those to whom you have provided the compiled version.

**Comment by carlosmn** [comment]
I gather the issue here is the question of whether doing a modification to libgit2 and then linking against that would still allow for the application using said library to remain proprietary (or under some other license). The modification in the linked issue seems to be conversion into webassembly, which to me seems too mechanical to become derivative work (and instead would... just be the same thing, I guess). But I am not a lawyer.

But as a senior software developer, I also get to say "it depends" a lot. The GPL works by placing restrictions on the distribution of "derived works" which is a thing in Copyright law. Ultimately a judge would be one to decide what happens. But the linking exception indicates that the authors do not consider using the library (and thus linking to it) to make a program so intrinsically bound to its functionality that it would cause the calling program to become a derived work. But whether something is a derived work doesn't necessarily hinge on how it's linked, but whether the work is, you know derived.

I'm going to stop myself before I go too deep into analysing what modification and distribution means over the last 30 years, and get back into what seemingly has caused the confusion that prompted this change.

The `(unmodified)` text goes all the way back so it wasn't in response to a particular earlier question (that I can tell). It reads to me to be trying to indicate that you can add libgit2 to your package manger and use it in your app and that's all good. Static or dynamic linking should not make a difference here (if nothing else this is evidenced by bindings from the same authors linking statically and having a different license).

But modifying libgit2 and then linking against that... that's probably fine, but it would depend on what the modifications are. What would it meant to make changes to expose internal mechanisms or utilities and then link against that? Unfortunately you'd have to say "it depends" because we don't know what those hypothetical modifications are. But in general I do agree that the word is more confusing than it should be.

I might reword it to something like

>`libgit2` is licensed under a **very permissive license** (GPLv2 with a special
Linking Exception).  This means that you can link against the library with any
kind of software without making that software fall under the GPL.
Changes to libgit2 would still be covered under its GPL license.
Additionally, the example code has been released to the public domain (see the
[separate license](examples/COPYING) for more information).

File: README.md (hunk 1 above)
Comment (ethomson) [reply README.md:29]: True.

**Comment by ethomson** [comment]
> I might reword it to something like
> 
> > libgit2 is licensed under a very permissive license (GPLv2 with a special
Linking Exception). This means that you can link against the library with any
kind of software without making that software fall under the GPL.
Changes to libgit2 would still be covered under its GPL license.
Additionally, the example code has been released to the public domain (see the
[separate license](https://github.com/libgit2/libgit2/pull/examples/COPYING) for more information).

✅  done

**Comment by liquidaty** [comment]
Thank you for making these changes. This is very helpful and greatly appreciated! Am hopeful I will have no more comments to pass on...
